### PR TITLE
test: rewrite tests for renaming network interface

### PR DESCRIPTION
**Pull request by yuwata**
- use `udevadm wait` instead of `udevadm info --wait-initialized`,
- use `timeout` command,
- add basic tests for #25106.

### CODE REVIEW

File: test/units/testsuite-17.02.sh [hunk 1]
```diff
@@ -1,59 +1,101 @@
...
+EOF
+
+udevadm control --log-priority=debug --reload --timeout=30
 
-SUBSYSTEM=="net", KERNEL=="lo", OPTIONS="log_level=debug"
-SUBSYSTEM=="net", KERNEL=="lo", TAG+="systemd", ENV{SYSTEMD_ALIAS}+="/sys/subsystem/net/devices/lo"
+ip link add hoge type dummy
+udevadm wait --timeout=30 --settle /sys/devices/virtual/net/hoge
+assert_not_in "ID_RENAMING=" "$(udevadm info /sys/devices/virtual/net/hoge)"
+timeout 30 bash -c 'while [[ "$(systemctl show --property=ActiveState --value /sys/devices/virtual/net/hoge)" != "active" ]]; do sleep .5; done'
```
Comment (keszybz) [inline test/units/testsuite-17.02.sh:35]: Not that it matters too much, but why `[[ ]]` instead of `[ ]`?

File: test/units/testsuite-17.02.sh [hunk 2]
```diff
@@ -1,59 +1,101 @@
...
-udevadm trigger --action=move --settle /sys/devices/virtual/net/lo
-udevadm info /sys/devices/virtual/net/lo
-sleep 1
-STATE=$(systemctl show --property=ActiveState --value sys-devices-virtual-net-lo.device)
-[[ $STATE == "active" ]] || exit 1
+udevadm control --log-priority=debug --reload --timeout=30
 
-rm -f /run/udev/rules.d/50-testsuite.rules
-udevadm control --reload --timeout=600
+# FIXME(?): the 'add' uevent is broadcasted as for 'foobar', instead of 'hoge'. Hence, we cannot use --settle here.
```
Comment (keszybz) [inline test/units/testsuite-17.02.sh]: s/broadcasted/broadcast/

**Review by keszybz — COMMENTED** [review]
Looks reasonable.

File: test/units/testsuite-17.02.sh (hunk 1 above)
Comment (yuwata) [reply test/units/testsuite-17.02.sh:35]: I have no idea about difference between them. Which is preferable way?

**Comment by yuwata** [comment]
Updated.
- Fixed comment,
- Silence shellcheck.

Upgrading the green label.

**Pull request merged**